### PR TITLE
Day4 class work and home work solutions

### PR DESCRIPTION
**Pull request by rathnakare**
Solutions for Day 4 class work and home work using multiple pointers

### CODE REVIEW

File: Java/Rathna/ReverseWordsInString.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        while(left < right) { //TC = O(n) and SC = O(1)
+            String temp = strings[left];
+            strings[left] = strings[right];
+            strings[right] = temp;
+            left++;
+            right--;
+        }
+        for(String s: strings) {
+            if(!s.isEmpty()) {
+                reverse += s + delimiter;
```
Comment (RahulKakarla) [inline Java/Rathna/ReverseWordsInString.java:25]: there will be an extra space at the end i think

File: Java/Rathna/Day4ClassWork.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import java.util.Arrays;
+
+public class Day4ClassWork {
```
Comment (RahulKakarla) [inline Java/Rathna/Day4ClassWork.java:3]: did you add time and space complexity somewhere

Comment (rathnakare) [reply Java/Rathna/Day4ClassWork.java:3]: as comments in the method

Comment (rathnakare) [reply Java/Rathna/Day4ClassWork.java:3]: missed for reversing an array

File: Java/Rathna/ReverseWordsInString.java (hunk 1 above)
Comment (rathnakare) [reply Java/Rathna/ReverseWordsInString.java:25]: yes, you are right. I will fix it

**Review by RahulKakarla — APPROVED** [review]

**Pull request merged**